### PR TITLE
JS: Add models for server-side form/file parsing libraries

### PR DESCRIPTION
**Pull request by erik-krogh**
[Evaluation](https://github.com/dsp-testing/erik-krogh-BCQS/tree/auto/data/workflow/parseForm-eval-1/reports) (and [a redo](https://github.com/dsp-testing/erik-krogh-BCQS/tree/auto/data/workflow/parseForm-eval-4/reports)) looks OK in terms of performance. 
4 new TPs. 
(The `js/tainted-path` results are TPs because an uploaded file can be named `..`).

### CODE REVIEW

**Comment by esbena** [comment]
What kind of false positive will we get by dropping the `and ... instanceof RequestExpr` conjuncts? My intuition is that the modeled  libraries always will use `RequestExpr`, regardless of our ability to detect that.

**Comment by erik-krogh** [comment]
> What kind of false positive will we get by dropping the and ... instanceof RequestExpr conjuncts? My intuition is that the modeled libraries always will use RequestExpr, regardless of our ability to detect that.

I have the same intuition.   

I suppose it's possible to give the libraries some other steam that just looks like a request object.   
But I don't expect that to be used outside of some internal test-cases.   
I haven't encountered such use in the wild. 

So I don't think we'll gain any false positives by dropping `and ... instanceof RequestExpr`.  
Do you think we should drop it?

**Comment by esbena** [comment]
> Do you think we should drop it?

Yes.

**Comment by erik-krogh** [comment]
> > Do you think we should drop it?
> 
> Yes.

Done.  
After removing `and ... instanceof RequestExpr` it was easy to refactor the code to use API-graphs, so I did that.

File: javascript/change-notes/2021-02-09-form-parsers.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+lgtm,codescanning
+* Server side form parsing libraries are now recognized as source of remote user input. 
+    [multer](https://www.npmjs.com/package/multer) and
+    [busboy](https://www.npmjs.com/package/busboy) and
+    [formidable](https://www.npmjs.com/package/formidable) and
+    [multiparty](https://www.npmjs.com/package/formidable)
```
Comment (asgerf) [inline javascript/change-notes/2021-02-09-form-parsers.md:6]:
```suggestion
  Affected packages are
    [multer](https://www.npmjs.com/package/multer),
    [busboy](https://www.npmjs.com/package/busboy),
    [formidable](https://www.npmjs.com/package/formidable), and
    [multiparty](https://www.npmjs.com/package/formidable).
```

**Review by asgerf — COMMENTED** [review]
LGTM, just the usual nit in the change note 😄

**Review by asgerf — APPROVED** [review]

**Pull request merged**